packages/safe-core-sdk-utils/src/eip-712/index.ts

```diff
@@ -68,5 +67,8 @@ export function generateTypedData({
       nonce: BigNumber.from(safeTransactionData.nonce)
     }
   }
+  if (eip712WithChainId) {
+    typedData.domain.chainId = chainId
```

Comment:
I guess the error was caused because ethers would try to encode `undefined` instead of ignoring it if the key is present?

Would it be possible to unit-test this? I feel like this is missed / refactored wrong easily.

Reply:
Yes, that was exactly the issue ;)
I'll add some tests to cover that